receiver/index.js

```diff
@@ -0,0 +1,191 @@
+
+    const userData = data.Users?.filter(
+      (user) =>
+        user.Attributes?.find((attr) => attr.Name == 'email')?.Value ==
+        userEmail
+    ).map((user) =>
+      user.Attributes?.filter((attr) =>
+        ['sub', 'email', 'name', 'custom:affiliation'].includes(attr.Name)
+      ).reduce(
+        (obj, item) => Object.assign(obj, { [item.Name]: item.Value }),
+        {}
+      )
+    )
+
+    if (userData.length == 0) {
+      await sendEmail(
+        userEmail,
+        'GCN Circular Submission Warning: Missing permissions',
+        'You do not have the required permissions to submit GCN Circulars. If you believe this to be a mistake, please fill out the form at https://heasarc.gsfc.nasa.gov/cgi-bin/Feedback?selected=kafkagcn, and we will look into resolving it as soon as possible.'
+      )
+      return
+    }
+
+    // Validate Subject Line
+    if (emailIsAutoReply(emailModel.subject)) {
+      return
+    }
+
+    if (!subjectIsValid(emailModel.subject)) {
+      await sendEmail(
+        userEmail,
+        'GCN Circular Submission Warning: Invalid subject structure',
+        'The submission of your Circular has been rejected, as the subject line does not conform to the appropriate format. Please see https://gcn.nasa.gov/circulars/classic#submission-process for more information.'
+      )
+      return
+    }
+
+    const circular = {
+      dummy: 0,
+      createdOn: Date.now(),
+      subject: emailModel.subject,
+      body: emailModel.text,
+      sub: userData[0].sub,
+      submitter: formatAuthor(userData[0]),
+    }
+
+    // Get tables for autoincrement
+    const command = new GetParametersCommand({
+      Names: DYNAMODB_TABLE_NAMES,
+    })
+    const response = await ssmClient.send(command)
+    if (!response) {
+      console.log('Throw an error here?')
+    }
+    const tableName = response.Parameters?.find(
+      (x) => x.Name == DYNAMODB_TABLE_NAMES[0]
+    )?.Value
+    const counterTableName = response.Parameters?.find(
+      (x) => x.Name == DYNAMODB_TABLE_NAMES[1]
+    )?.Value
+
+    if (!tableName || !counterTableName) {
+      console.log('Nope, didnt find the names, throw an error')
+    }
+
+    const doc = DynamoDBDocument.from(dynamodbClient)
+    const autoIncrement = new DynamoDBAutoIncrement({
+      doc,
+      counterTableName,
+      counterTableKey: { tableName: 'circulars' },
+      counterTableAttributeName: 'circularId',
+      tableName: tableName,
+      tableAttributeName: 'circularId',
+      initialValue: 1,
+    })
+    const newCircularId = await autoIncrement.put(circular)
+
+    // Send a success email
+    await sendEmail(
+      userEmail,
+      'Successfully submitted Circular: ' + newCircularId,
+      'Your circular has been successfully submitted. You may view it at http://gcn.nasa.gov/circulars/' +
+        newCircularId
+    )
+
+    return 'Success'
+  } catch (err) {
+    console.log(err)
+
+    throw err
+  }
```

Comment:
Why do we need to catch and re-throw the error?

Reply:
We don't, this was just left over from prior testing. Removing now